satpy/readers/seviri_l1b_icare.py

```diff
@@ -259,7 +262,9 @@ def get_dataset(self, ds_id, ds_info):
             # Now we correct range from 0-1 to 0-100 for VIS:
             if ds_id['name'] in self.ref_bands:
                 data.values *= 100.
-        return data
+        variable = xr.DataArray(da.from_array(data, chunks=(CHUNK_SIZE, CHUNK_SIZE)),
+                                dims=['y', 'x'])
```

Comment:
I think the problem comes from a few lines above with: `data.values = data.values.astype(np.float32)`. This looks like it converts the daskified DataArray to numpy. Doesn't `data = data.astype(np.float32)` work instead?

Reply:
No, that doesn't work as the data returned from the HDF4 file is a numpy array, it has to be explicitly converted to dask / xarray.
That said, I didn't need `.values` in the code so have at least deleted that :)